collector/tcpstat_linux.go

```diff
@@ -73,7 +81,7 @@ func NewTCPStatCollector(logger *slog.Logger) (Collector, error) {
 		desc: typedDesc{prometheus.NewDesc(
 			prometheus.BuildFQName(namespace, "tcp", "connection_states"),
 			"Number of connection states.",
-			[]string{"state"}, nil,
+			[]string{"state", "port", "type"}, nil,
```

Comment:
These labels need to be optional for the case that the user has not specified any filtering.

Also `type` is not a good label name, it's too generic. Type of what?